tasks/main.yml

```diff
@@ -47,6 +50,22 @@
     value: "{{ public_ip.ansible_facts.ipify_public_ip }}"
     aws_access_key: "{{ update_ip_r53_aws_access_key }}"
     aws_secret_key: "{{ update_ip_r53_aws_secret_key }}"
-    wait: "{{ item.wait | default('yes') }}"
-    overwrite: yes
+    wait: "{{ item.wait | default('true') }}"
+    overwrite: true
   with_items: "{{ update_ip_r53_records }}"
+  when: not use_public_ip | bool
+
+- name: Update the DNS record(s)
```

Comment:
This is essentially the same as the above task except the `value` field is different. After removing the default `private_ip` value as requested, the above value filed could be changed to something like `value: "{{ private_ip | default(public_ip.ansible_facts.ipify_public_ip) }}"`.